MIDAS/src/systems.cpp

```diff
@@ -28,9 +28,27 @@ DECLARE_THREAD(logger, RocketSystems* arg) {
 }
 
 DECLARE_THREAD(barometer, RocketSystems* arg) {
+    // Reject single rogue barometer readings that are very different from the immediately prior reading
+    // Will only reject a certain number of readings in a row
+    Barometer prev_reading;
+    constexpr float altChgThreshold = 200; // meters
+    constexpr float presChgThreshold = 500; // milibars
+    constexpr float tempChgThreshold = 10; // degrees C
+    constexpr unsigned int maxConsecutiveRejects = 5;
+    unsigned int rejects = maxConsecutiveRejects; // Always accept first reading
     while (true) {
         Barometer reading = arg->sensors.barometer.read();
-        arg->rocket_data.barometer.update(reading);
+        bool is_rogue = std::abs(prev_reading.altitude - reading.altitude) > altChgThreshold ||
```

Comment:
Do we need to check all three? Altitude is a function of both temperature and pressure, so any issues with temp and pressure will propagate through to altitude. Threshold changes are harder to conceptualize with pressure, so it may be harder to determine an accurate pressure threshold unless we hitl. I would say only checking altitude would be enough, but if we adequately test to determine threshold values, we can keep all 3 to be safe.